identity-service/src/relay/ethTxRelay.js

```diff
@@ -0,0 +1,217 @@
+    req.logger.info(`L1 txRelay - success, req:${reqBodySHA}, sender:${senderAddress}`)
+    return txHash
+}
+
+const createAndSendEthTransaction = async (sender, receiverAddress, value, web3, logger, gasPrice, gasLimit = null, data = null) => {
+    const privateKeyBuffer = Buffer.from(sender.privateKey, 'hex')
+    const walletAddress = EthereumWallet.fromPrivateKey(privateKeyBuffer)
+    const address = walletAddress.getAddressString()
+    if (address !== sender.publicKey.toLowerCase()) {
+        throw new Error(`Invalid relayerPublicKey found. Expected ${sender.publicKey.toLowerCase()}, found ${address}`)
+    }
+
+    const nonce = await web3.eth.getTransactionCount(address)
+    let txParams = {
+        nonce: web3.utils.toHex(nonce),
+        gasPrice,
+        gasLimit: gasLimit ? web3.utils.numberToHex(gasLimit) : DEFAULT_GAS_LIMIT,
+        to: receiverAddress,
+        value: web3.utils.toHex(value)
+    }
+    logger.info(`Final params: ${JSON.stringify(txParams)}`)
+
+    if (data) {
+        txParams = { ...txParams, data }
+    }
+
+    const tx = new EthereumTx(txParams)
+    tx.sign(privateKeyBuffer)
+
+    const signedTx = '0x' + tx.serialize().toString('hex')
+
+    logger.info(`rawGasPrice: ${gasPrice}`)
+    logger.info(`txRelay - sending a transaction for sender ${sender.publicKey} to ${receiverAddress}, gasPrice ${parseInt(gasPrice, 16)}, gasLimit ${DEFAULT_GAS_LIMIT}, nonce ${nonce}`)
+    // const receipt = await web3.eth.sendSignedTransaction(signedTx)
+    const txHash = await sendSignedTransactionReturnOnTxHash(web3, signedTx, logger)
+
+    return { txHash, txParams }
+}
+
+const sendSignedTransactionReturnOnTxHash = async (web3, signedTx, logger) => {
+  return new Promise((resolve, reject) => {
+    try {
+      web3.eth.sendSignedTransaction(signedTx)
+        .once('transactionHash', function(hash){
+          // Resolve this promise with a tx hash has been returned
+          resolve(hash);
+        })
+        .once('receipt', function(receipt){ logger.info(`Receipt returned ${JSON.stringify(receipt)}`)  })
+        .on('confirmation', function(confNumber, receipt){ logger.info(`${JSON.stringify(confNumber)}, ${JSON.stringify(receipt)}`)   })
+        .on('error', function(error){ logger.error(JSON.stringify(error))  })
+        .then(function(receipt){
+          logger.info(`Success! Processed ${JSON.stringify(receipt)}`)
+            // will be fired once the receipt is mined
+        });
+    } catch (err) {
+        reject(err);
+    }
+  })
+}
+
+// Query mainnet ethereum gas prices
+const getProdGasInfo = async (redis, logger) => {
+    const prodGasPriceKey = 'eth-gas-prod-price-info'
+    let gasInfo = await redis.get(prodGasPriceKey)
+    if (!gasInfo) {
+      logger.info(`Redis cache miss, querying remote`)
+      let prodGasInfo = await axios({
+        method: 'get',
+        url: 'https://ethgasstation.info/api/ethgasAPI.json'
```

Comment:
consider purchasing an api key for this and also having a fallback eth specific gas price in case we either don't have a cached value